### PR TITLE
Upgrade to ucx 1.11 and remove unnecessary patches

### PR DESCRIPTION
**Pull request by quasiben**
UCX 1.11 was officially tagged [today](https://github.com/openucx/ucx/releases/tag/v1.11.0) and this PR updates the conda package for building 1.11

Note: UCX 1.11 no longer needs any patches

cc @jakirkham @pentschev

### CODE REVIEW

**Comment by pentschev** [comment]
We still have one pending bug https://github.com/rapidsai/dask-cuda/issues/677 that is going to be fixed by https://github.com/openucx/ucx/pull/7140 and will only be in 1.11.1. I would rather skip this release again and have one that solves all our issues instead.

**Review by pentschev — CHANGES_REQUESTED** [review]
For now I'm blocking this. Unless there's a very good reason for which we want 1.11.0, I would prefer that we skip to 1.11.1, since we're also not using it for 21.08.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,8 +23,6 @@ source:
   - git_url: https://github.com/openucx/ucx
     git_rev: {{ ucx_commit }}
     folder: ucx
-    patches:
-      - cuda-alloc-rcache.patch  # [cuda_compiler_version != "None"]
```
Comment (jakirkham) [inline recipe/meta.yaml:27]: Can we please delete the patch file itself as well?

https://github.com/rapidsai/ucx-split-feedstock/blob/master/recipe/cuda-alloc-rcache.patch

**Comment by jakirkham** [comment]
Would it make sense to just include that as a patch for now Peter? Not sure how badly people need 1.11 atm

**Comment by quasiben** [comment]
Unfortunately, this was premature.  1.11.1 will have a critical fix -- I'll close this PR and reopen a new one when 1.11.1 is available

**Comment by jakirkham** [comment]
Ok sounds good :)

File: recipe/meta.yaml (hunk 1 above)
Comment (pentschev) [reply recipe/meta.yaml:27]: For future reference, deleting the file should be ok too. [Where users may still want it, we reference it by the commit ID](https://ucx-py.readthedocs.io/en/latest/install.html#ucx-1-9).

**Comment by pentschev** [comment]
> Would it make sense to just include that as a patch for now Peter? Not sure how badly people need 1.11 atm

Exactly because of that, I prefer skipping 1.11.0 and waiting another week or so for 1.11.1. :)

**Pull request closed without merging**